From 07271aea68e2ab68affec36cce23c3870878cae1 Mon Sep 17 00:00:00 2001
From: Piotr Tworek <ptworek@vewd.com>
Date: Thu, 01 Oct 2020 09:19:03 +0000
Subject: [PATCH] Fix partition_alloc build issues with libstdc++.

Two minor and pretty generic issues are exposed when building the code
with libstdc++ instead of bundled libcxx.

1. Code in thread_cache.h uses std::atomic but fails to include required
   header file.
2. Code in partition_alloc_constants.h uses std::max_align_t which is
   defined in cstddef, not its C version stddef.h.

In both cases libcxx basically pulls in required headers indirectly.

Bug: 957519
Change-Id: Idcee3e7abfab94cb40e1b95c636b97cf9c766b75
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/2440436
Reviewed-by: Kentaro Hara <haraken@chromium.org>
Commit-Queue: Piotr Tworek <ptworek@vewd.com>
Cr-Commit-Position: refs/heads/master@{#812598}
---

diff --git a/base/allocator/partition_allocator/partition_alloc_constants.h b/base/allocator/partition_allocator/partition_alloc_constants.h
index b295cab..e332140 100644
--- a/base/allocator/partition_allocator/partition_alloc_constants.h
+++ b/base/allocator/partition_allocator/partition_alloc_constants.h
@@ -6,7 +6,7 @@
 #define BASE_ALLOCATOR_PARTITION_ALLOCATOR_PARTITION_ALLOC_CONSTANTS_H_
 
 #include <limits.h>
-#include <stddef.h>
+#include <cstddef>
 
 #include <algorithm>
 
diff --git a/base/allocator/partition_allocator/thread_cache.h b/base/allocator/partition_allocator/thread_cache.h
index dd71ccc..0509747 100644
--- a/base/allocator/partition_allocator/thread_cache.h
+++ b/base/allocator/partition_allocator/thread_cache.h
@@ -5,6 +5,7 @@
 #ifndef BASE_ALLOCATOR_PARTITION_ALLOCATOR_THREAD_CACHE_H_
 #define BASE_ALLOCATOR_PARTITION_ALLOCATOR_THREAD_CACHE_H_
 
+#include <atomic>
 #include <cstdint>
 #include <memory>
 
